mlx_graphs/loaders/neighbor_loader.py

```diff
@@ -0,0 +1,44 @@
+from typing import List
+
+import mlx.core as mx
+import numpy as np
+
+from mlx_graphs.data import GraphData
+from mlx_graphs.sampler.neighbor_sampler import sample_nodes
+
+
+class NeighborLoader:
```

Comment:
I'm not sure whether this should be a data loader or just a sampler that we can use to generate samples to be loaded through the current `Dataloader`. @TristanBilot @Kheims what do you think?

Reply:
Hmm, imo you're right we should directly use the DataLoader, but we should therefore provide to the data loader a sampler function which means some modifications needs to done in it ?

Reply:
It can also be a two-stage process, like:
```python
graph = GraphData(...)
samples = sampler(graph, num_samples)
dl = Dataloader(samples, batch_size)
```

Reply:
Seems good to me

Reply:
Interesting. Yeah I find that using a dedicated `DataLoader` everytime we want to sample is suboptimal. We should provide both a `sample` function and a `NeighborSampler` class which can be added as argument to every `DataLoader`. 
However, we have to check also the compatibility between the graph batching of the `DataLoader` and the node batching of the `NeighborSampler`.

Reply:
It's the same approach of [GraphSAGE](https://arxiv.org/pdf/1706.02216.pdf), right?

Reply:
Yes absolutely, the question is just to make sure about the implementation details (not cited in the original paper but done in implementations of pyg and dgl ( but they rely on an cpp implementation))

Reply:
Yes, that seems correct to me - you can create a dummy test with a graph like
```python
#   0  1
#  / \/ \
#  2  3  4
#  |  |  |
#  5  6  7
```
where the sampling should be deterministic (see the [example in pyg](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/neighbor_loader.html)).

Reply:
In the example they use a batch size of 1 where we expect to have dissociated subgraphs for each input nodes, but if batch size is 2 with input nodes 0 and 1 then we should associate the subgraphs to contain a single computational graph containing both subgraphs ( in this case all the graph you mentioned ) 
This is a straightforward example but i wanted a more complex one to test robustly the functionality ( where we had choices in sampling in each hop )

Reply:
@Kheims I'm not sure I understand - when batch size is grater than 1, shouldn't you just repeat the sampling procedure and then collate the resulting subgraphs?

Reply:
@Kheims I think you're right - my understanding is that the procedure you were suggesting is the correct one, rather than just sampling and collating the results. @TristanBilot do you agree?